internal/template/character/stam.go

```diff
@@ -102,18 +111,51 @@ func (c *Character) DashLength() int {
 	}
 }
 
-func (c *Character) Jump(p map[string]int) (action.Info, error) {
-	f := 30
+func (c *Character) DashToJumpLength() int {
 	switch c.CharBody {
-	case info.BodyBoy, info.BodyGirl:
-		f = 31
-	case info.BodyMale:
-		f = 28
-	case info.BodyLady:
-		f = 32
-	case info.BodyLoli:
-		f = 29
+	case info.BodyGirl, info.BodyLoli:
+		return 4
+	case info.BodyMale, info.BodyLady:
+		return 3
+	case info.BodyBoy:
+		return 2
+	default:
+		return 3
 	}
+}
+
+func (c *Character) Jump(p map[string]int) (action.Info, error) {
+	if c.StatusIsActive(player.XianyunAirborneBuff) {
+		c.Core.Player.SetAirborne(player.AirborneXianyun)
+		// 4/8 for claymore/bow/catalyst and 5/9 for sword/polearm
+		lowPlunge := 4
+		switch c.Weapon.Class {
+		case info.WeaponClassSword, info.WeaponClassSpear:
+			lowPlunge = 5
+		}
+
+		highPlunge := 8
+		switch c.Weapon.Class {
+		case info.WeaponClassSword, info.WeaponClassSpear:
+			highPlunge = 9
+		}
```

Comment:
Please combine these two switch statements because you don't need to check the weapon class twice in the same way and to save vertical space in an already long if block, so
```
// 4/8 for claymore/bow/catalyst and 5/9 for sword/polearm
lowPlunge := 4
highPlunge := 8
switch c.Weapon.Class {
case info.WeaponClassSword, info.WeaponClassSpear:
	lowPlunge = 5
	highPlunge = 9
}
```